sdk/cdn/Microsoft.Azure.Management.Cdn/src/Generated/Models/CookiesMatchConditionParameters.cs

```diff
@@ -34,17 +34,17 @@ public CookiesMatchConditionParameters()
         /// Initializes a new instance of the CookiesMatchConditionParameters
         /// class.
         /// </summary>
-        /// <param name="selector">Name of Cookies to be matched</param>
         /// <param name="operatorProperty">Describes operator to be matched.
         /// Possible values include: 'Any', 'Equal', 'Contains', 'BeginsWith',
         /// 'EndsWith', 'LessThan', 'LessThanOrEqual', 'GreaterThan',
         /// 'GreaterThanOrEqual'</param>
-        /// <param name="matchValues">The match value for the condition of the
-        /// delivery rule</param>
+        /// <param name="selector">Name of Cookies to be matched</param>
         /// <param name="negateCondition">Describes if this is negate condition
         /// or not</param>
+        /// <param name="matchValues">The match value for the condition of the
+        /// delivery rule</param>
         /// <param name="transforms">List of transforms</param>
-        public CookiesMatchConditionParameters(string selector, string operatorProperty, IList<string> matchValues, bool? negateCondition = default(bool?), IList<string> transforms = default(IList<string>))
+        public CookiesMatchConditionParameters(string operatorProperty, string selector = default(string), bool? negateCondition = default(bool?), IList<string> matchValues = default(IList<string>), IList<string> transforms = default(IList<string>))
         {
```

Comment:
This appears to a be a breaking change. A caller passing in string, string, Ilist, bool into this will now fail since the 3rd parameter is now a bool and not an IList.

Reply:
Ahh, you're right. Unfortunately it's a bug that needs to be fixed, and this wasn't a breaking change for the REST API specs since REST parameters aren't positional, so it has already been accepted there. What's the procedure for getting a breaking change released?

Reply:
Swapping the params around in swagger ought to fix it, since you are just swapping around optional params.

Since swagger is merged, may be able to do some customization.

Customization options are here:

https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview

Reply:
I've done some work trying to use a customization to correct this issue, but it seems that only data plane libraries are currently using customizations, and I'm having a difficult time referencing Azure.Core like in the example you provided as required by the autorest .Net code generator for removing the backwards-incompatible constructor.

Are customizations of generated code supported for management modules? Adding Azure.Core requires different .Net versions than the management modules are supposed to reference, as stated in CONTRIBUTING.md.

Reply:
The example I gave of DNS is management, although it is track 2.  

I think all that may need done is to make a file call /Customization/Models/CookiesMatchConditionParameters.cs

And then in there, change CookiesMatchConditionParameters to:

public CookiesMatchConditionParameters(string selector, string selector = default(string), IList<string> matchValues = default(IList<string>), bool? negateCondition = default(bool?), IList<string> transforms = default(IList<string>))

Reply:
Ahh I see, the difference between track 2 management and data plane wasn't quite clear from the doc I read, makes sense though.

Your solution will work as long as you're ok with having redundant constructors - The issue with importing Azure.Core only applies for deleting the non backwards compatible constructor. I'll push the change with both constructors, just let me know if that isn't acceptable.